### PR TITLE
Added an addon for Thanos

### PR DESCRIPTION
**Pull request by aayvazyan**
**What type of PR is this?**
Feature

**What this PR does/ why we need it**:
This PR adds a new `Thanos` addon which uses upstream Thanos chart from https://github.com/banzaicloud/banzai-charts

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-65810

**Special notes for your reviewer**:
The Thanos addon is tested and can be enabled for any konvoy cluster having the following extra configurations and prerequisites.

##### Prerequisites
1. The nodes in the konvoy cluster have write access to the corresponding AWS S3 bucket, in this example we chose `your-thanos-s3-bucket` as S3 bucket name. 

2. A k8s secret configured _for example_ in **_./extras/kubernetes/thanos-config.yaml_** which contains the S3 configuration for `sidecar`

```
apiVersion: v1
kind: Secret
metadata:
  name: thanos-objstore-config
  namespace: kubeaddons
type: Opaque
stringData:
  object-store.yaml: |-
    type: s3
    config:
      bucket: your-thanos-s3-bucket
      endpoint: s3.us-west-2.amazonaws.com
```
3. Konvoy **cluster.yaml** should contain the following sections 

##### Thanos enabled from the [`kubernetes-sre-addons`](https://github.com/mesosphere/kubernetes-sre-addons)

```
  - configRepository: https://github.com/mesosphere/kubernetes-sre-addons
    configVersion: "sre/thanos-0.3.x"
    addonsList:
    - name: thanos
      enabled: true
      values: |
        objstoreSecretOverride: "thanos-objstore-config"
```

##### Prometheus configured with enabled `sidecar` and Grafana having Thanos as additional data source.

```
    - name: prometheus
      enabled: true
      values: | 
        grafana:
          sidecar:
            datasources:
              enabled: true
              defaultDatasourceEnabled: false
          additionalDataSources:
            - name: thanos-query
              access: proxy
              type: prometheus
              url: "http://thanos-kubeaddons-query-http:10902"
        prometheus:
          prometheusSpec:
            thanos:
              objectStorageConfig:
                key: object-store.yaml
                name: thanos-objstore-config
```

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [x] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: addons/thanos/0.3.x/thanos-1.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      compact:
+        enabled: false
+      sidecar:
+        enabled: false
+      query:
+        enabled: true
+        extraArgs:
+          - "--store=prometheus-kubeaddons-prom-prometheus.kubeaddons.svc:10901"
+          - "--query.auto-downsampling"
+          - "--query.partial-response"
```
Comment (sebbrandt87) [inline addons/thanos/0.3.x/thanos-1.yaml]: missing linebreak

File: addons/thanos/0.3.x/thanos-1.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    minSupportedVersion: v1.15.6
+  cloudProvider:
+    - name: aws
+      enabled: true
+    - name: azure
+      enabled: true
+    - name: docker
+      enabled: false
+    - name: none
+      enabled: true
```
Comment (sebbrandt87) [inline addons/thanos/0.3.x/thanos-1.yaml]:
```suggestion
  cloudProvider:
    - name: aws
      enabled: true
    - name: azure
      enabled: false
    - name: gcp
      enabled: false
    - name: docker
      enabled: false
    - name: none
      enabled: false
```

**Review by sebbrandt87 — CHANGES_REQUESTED** [review]

File: addons/thanos/0.3.x/thanos-1.yaml (hunk 2 above)
Comment (sebbrandt87) [reply addons/thanos/0.3.x/thanos-1.yaml]: maybe we should disable it for all others except for `aws` as long as we do not provide permission / prerequisites details for `azure`, `gcp` and `none`

Comment (aayvazyan) [reply addons/thanos/0.3.x/thanos-1.yaml]: Thanos has plenty of backends [documented](https://github.com/thanos-io/thanos/blob/master/docs/storage.md#configuration), so let me disagree here that we need to disable all other cloud providers ;)
I'll paste the link with the docs into [our usage example doc](https://github.com/mesosphere/sre-konvoy-deployers/pull/22/commits/6a969bc002d973846b16964bcec17dca247c6cab)

**Review by sebbrandt87 — APPROVED** [review]

File: addons/thanos/0.3.x/thanos-1.yaml (hunk 2 above)
Comment (sebbrandt87) [reply addons/thanos/0.3.x/thanos-1.yaml]: but would you please add then `gcp` 😬

**Comment by aayvazyan** [comment]
Thanks for the review @sebbrandt87! :) 
Merging this.

**Pull request merged**